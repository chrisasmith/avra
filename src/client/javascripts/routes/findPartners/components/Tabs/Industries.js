import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { injectIntl, defineMessages } from 'react-intl'
import styles from '../styles/FindPartners.scss'
import * as findPartnersActions from '../../../../reducers/pages/findPartnersActions'
import { createLoadingSelector } from '../../../../reducers/selectors'
import GridCard, { CARD_TYPE } from '../../../../components/GridCards/GridCard'
import FilterBreadCrumb from '../../../../components/Filter/FilterBreadCrumb'
import IndustryBreadcrumb from './components/IndustryBreadcrumb'

const propTypes = {
  data: PropTypes.object,
  filter: PropTypes.object,
  params: PropTypes.object.isRequired,
  isFetching: PropTypes.bool,
  intl: PropTypes.object,
}

const messages = defineMessages({
  topProducts: {
    id: 'find_partners.tab.industries.top_products',
    defaultMessage: 'Top Products',
  },
  productCurrentlyOwn: {
    id: 'find_partners.tab.industries.product_currently_own',
    defaultMessage: 'Which of the following products do you currently own or have you bought recently?',
  },
  autoOwnership: {
    id: 'find_partners.tab.industries.auto_ownership',
    defaultMessage: 'Auto Ownership',
  },
  autoOwnershipDesc: {
    id: 'find_partners.tab.industries.auto_ownership_desc',
    defaultMessage: 'Make of any vehicle owned or leased',
  },
  beerConsumed30: {
    id: 'find_partners.tab.industries.beer_consumed_30',
    defaultMessage: 'Beer Consumed in Last 30 Days',
  },
  softDrinks: {
    id: 'find_partners.tab.industries.soft_drinks',
    defaultMessage: 'Soft Drinks Consumed in Last 7 Days',
  },
  telephoneServiceProvider: {
    id: 'find_partners.tab.industries.telephone_service_providers',
    defaultMessage: 'Telephone Service Providers',
  },
  wirelessPhoneCarriers: {
    id: 'find_partners.tab.industries.wireless_phone_carriers',
    defaultMessage: 'Current Wireless/Cell Phone Carriers',
  },
  internetService: {
    id: 'find_partners.tab.industries.internet_services',
    defaultMessage: 'Internet Service Providers',
  },
  cableTv: {
    id: 'find_partners.tab.industries.cable_tv',
    defaultMessage: 'Cable/TV Service Provider',
  },
})

const IndustriesTab = injectIntl(({ intl, filter, data, isFetching, params }) => {
  const industry = data ? data.industries.find(i => i.id === params.industry_id) : null

  const industryId = params.industry_id
  let subsection = null

  if (industry && industry.subsections && !params.subsection_id) {
    subsection = industry.subsections.sort((a, b) => a.name.localeCompare(b.name))[0]
  } else if (industry && industry.subsections && params.subsection_id) {
    subsection = industry.subsections.find(s => s.id === params.subsection_id)
  }

  const title = subsection ? subsection.name : industry ? industry.name : ''
  const subsectionId = subsection ? subsection.id : null

  const leftColumn = [
    {
      type: CARD_TYPE.INDEX,
      dataKey: 'top_partners',
      cardData: {
        title: 'Top Products',
        description: 'Which of the following products do you currently own or have you bought recently?',
      },
    },
  ]

  const centerColumn = [
    {
      type: CARD_TYPE.STRING,
      cardData: {
        title: `${title} Overview`,
        value: 'Placeholder',
      },
    },
  ]

  const rightColumn = [
    {
      hide: industryId !== 'auto',
      type: CARD_TYPE.INDEX,
      dataKey: 'auto_ownership',
      cardData: {
        title: intl.formatMessage(messages.autoOwnership),
        description: intl.formatMessage(messages.autoOwnershipDesc),
        scarborough: true,
      },
    }, {
      hide: subsectionId !== 'beer',
      type: CARD_TYPE.INDEX,
      dataKey: 'beer',
      cardData: {
        scarborough: true,
        title: intl.formatMessage(messages.beerConsumed30),
      },
    }, {
      hide: subsectionId !== 'soft_drinks',
      type: CARD_TYPE.INDEX,
      dataKey: 'soft_drinks',
      cardData: {
        scarborough: true,
        title: intl.formatMessage(messages.softDrinks),
      },
    }, {
      hide: subsectionId !== 'telephone',
      type: CARD_TYPE.INDEX,
      dataKey: 'telephone_service_providers',
      cardData: {
        title: intl.formatMessage(messages.telephoneServiceProvider),
        scarborough: true,
      },
    }, {
      hide: subsectionId !== 'wireless_phone_carriers',
      type: CARD_TYPE.INDEX,
      dataKey: 'wireless_phone_carriers',
      cardData: {
        scarborough: true,
        title: intl.formatMessage(messages.wirelessPhoneCarriers),
      },
    }, {
      hide: subsectionId !== 'internet',
      type: CARD_TYPE.INDEX,
      dataKey: 'internet_services',
      cardData: {
        scarborough: true,
        title: intl.formatMessage(messages.internetService),
      },
    }, {
      hide: subsectionId !== 'cable_tv',
      type: CARD_TYPE.INDEX,
      dataKey: 'cable_tv',
      cardData: {
        scarborough: true,
        title: intl.formatMessage(messages.cableTv),
      },
    },
  ]

  return (
    <div className={styles.grid}>
      <div>
        <IndustryBreadcrumb industry={industry} params={params} />
      </div>
      <div className={styles.breadCrumb}>
        <FilterBreadCrumb filter={filter} />
      </div>
      <div className="container">
        <div className="row row-grid">
          <div className="col-12">
            {centerColumn.map((card, idx) => (
              <GridCard key={idx} card={card} data={data} isFetching={isFetching} />
            ))}
          </div>
          <div className="col-6">
            {leftColumn.map((card, idx) => (
              <GridCard key={idx} card={card} data={data} isFetching={isFetching} />
            ))}
          </div>
          <div className="col-6">
            {rightColumn.map((card, idx) => (
              <GridCard key={idx} card={card} data={data} isFetching={isFetching} />
            ))}
          </div>
        </div>
      </div>
    </div>
  )
})

function mapStateToProps(state) {
  const loadingSelector = createLoadingSelector([findPartnersActions.GET_FIND_PARTNERS_PAGE.SELF])
  const filter = state.filters.filter
  const data = state.pages.findPartners.data

  return { ...state, isFetching: loadingSelector(state), filter, data }
}

IndustriesTab.propTypes = propTypes
export default connect(mapStateToProps)(IndustriesTab)
