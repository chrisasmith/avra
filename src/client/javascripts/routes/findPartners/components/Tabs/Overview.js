import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { injectIntl, defineMessages } from 'react-intl'
import styles from '../styles/FindPartners.scss'
import * as findPartnersActions from '../../../../reducers/pages/findPartnersActions'
import { createLoadingSelector } from '../../../../reducers/selectors'
import GridCard, { CARD_TYPE } from '../../../../components/GridCards/GridCard'
import FilterBreadCrumb from '../../../../components/Filter/FilterBreadCrumb'

const propTypes = {
  data: PropTypes.object,
  filter: PropTypes.object,
  isFetching: PropTypes.bool,
  intl: PropTypes.object,
}

const messages = defineMessages({
  topProducts: {
    id: 'find_partners.tab.overview.top_products',
    defaultMessage: 'Top Products',
  },
  productCurrentlyOwn: {
    id: 'find_partners.tab.overview.currently_own',
    defaultMessage: 'Which of the following products do you currently own or have you bought recently?',
  },
  topServices: {
    id: 'find_partners.tab.overview.top_services',
    defaultMessage: 'Top Services',
  },
  serviceCurrentlyUse: {
    id: 'find_partners.tab.overview.service_currently_use',
    defaultMessage: 'Which of the following services do you currently use or have you used in the last 12 months? And which do you intend to use again in the next 12 months?',
  },
})

const OverviewTab = injectIntl(({ intl, filter, data, isFetching }) => {
  const leftColumn = [
    {
      type: CARD_TYPE.INDEX,
      dataKey: 'top_partners',
      cardData: {
        title: intl.formatMessage(messages.topProducts),
        description: intl.formatMessage(messages.productCurrentlyOwn),
      },
    },
  ]

  const rightColumn = [
    {
      type: CARD_TYPE.INDEX,
      dataKey: 'top_services',
      cardData: {
        title: intl.formatMessage(messages.topServices),
        description: intl.formatMessage(messages.serviceCurrentlyUse),
      },
    },
  ]

  return (
    <div className={styles.grid}>
      <div className={styles.breadCrumb}>
        <FilterBreadCrumb filter={filter} />
      </div>
      <div className="container">
        <div className="row row-grid">
          <div className="col-6">
            {leftColumn.map((card, idx) => (
              <GridCard key={idx} card={card} isFetching={isFetching} data={data} />
            ))}
          </div>
          <div className="col-6">
            {rightColumn.map((card, idx) => (
              <GridCard key={idx} card={card} isFetching={isFetching} data={data} />
            ))}
          </div>
        </div>
      </div>
    </div>
  )
})

function mapStateToProps(state) {
  const loadingSelector = createLoadingSelector([findPartnersActions.GET_FIND_PARTNERS_PAGE.SELF])
  const filter = state.filters.filter
  const data = state.pages.findPartners.data
  return { ...state, isFetching: loadingSelector(state), filter, data }
}

OverviewTab.propTypes = propTypes
export default connect(mapStateToProps)(OverviewTab)
