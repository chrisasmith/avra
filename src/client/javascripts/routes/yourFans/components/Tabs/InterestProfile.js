import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { injectIntl, defineMessages } from 'react-intl'
import styles from '../styles/YourFans.scss'
import * as yourFansActions from '../../../../reducers/pages/yourFansActions'
import { createLoadingSelector } from '../../../../reducers/selectors'
import DashboardGraphModule, { MODULE_TYPE } from '../../../../components/GridCards/DashboardGraphModule'
import FilterBreadCrumb from '../../../../components/Filter/FilterBreadCrumb'

const propTypes = {
  data: PropTypes.object,
  filter: PropTypes.object,
  isFetching: PropTypes.bool.isRequired,
  intl: PropTypes.object,
}

const messages = defineMessages({
  areasTitle: {
    id: 'your_fans.tab.interest.areas_title',
    defaultMessage: 'Interest Areas',
  },
  areasSideTitle: {
    id: 'your_fans.tab.interest.areas_side',
    defaultMessage: 'Interest areas',
  },
  additionalInterest: {
    id: 'your_fans.tab.interest.additional',
    defaultMessage: 'Additional Interests',
  },
  sportsInterestArea: {
    id: 'your_fans.tab.interest.sports_area',
    defaultMessage: 'Sports Interest Areas',
  },
})

const InterestProfileTab = injectIntl(({ intl, filter, data, isFetching }) => (
  <div className={styles.grid}>
    <div className={styles.breadCrumb}>
      <FilterBreadCrumb filter={filter} />
    </div>
    <div className="container">
      <div className="row row-grid">
        <div className="col-6">
          <DashboardGraphModule
            title={intl.formatMessage(messages.areasTitle)}
            sideTitle={intl.formatMessage(messages.areasSideTitle)}
            data={data ? data.interest_areas : null}
            isFetching={isFetching}
            type={MODULE_TYPE.PERCENT}
            propertyTitle={filter.property.name}
            benchmarkTitle={filter.benchmark.name}
          />
          <DashboardGraphModule
            title={intl.formatMessage(messages.additionalInterest)}
            sideTitle={intl.formatMessage(messages.areasSideTitle)}
            data={data ? data.additional_interests : null}
            isFetching={isFetching}
            type={MODULE_TYPE.PERCENT}
            propertyTitle={filter.property.name}
            benchmarkTitle={filter.benchmark.name}
          />
        </div>
        <div className="col-6">
          <DashboardGraphModule
            title={intl.formatMessage(messages.sportsInterestArea)}
            sideTitle={intl.formatMessage(messages.areasSideTitle)}
            data={data ? data.sports_interest_areas : null}
            isFetching={isFetching}
            type={MODULE_TYPE.PERCENT}
            propertyTitle={filter.property.name}
            benchmarkTitle={filter.benchmark.name}
          />
        </div>
      </div>
    </div>
  </div>
))

function mapStateToProps(state) {
  const loadingSelector = createLoadingSelector([yourFansActions.GET_HOME_PAGE.SELF])
  const filter = state.filters.filter
  const data = state.pages.yourFans.data
  return { ...state, isFetching: loadingSelector(state), filter, data }
}

InterestProfileTab.propTypes = propTypes
export default connect(mapStateToProps)(InterestProfileTab)
