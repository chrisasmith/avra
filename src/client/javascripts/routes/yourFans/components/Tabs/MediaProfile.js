import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { injectIntl, defineMessages } from 'react-intl'
import styles from '../styles/YourFans.scss'
import * as yourFansActions from '../../../../reducers/pages/yourFansActions'
import { createLoadingSelector } from '../../../../reducers/selectors'
import DashboardGraphModule, { MODULE_TYPE } from '../../../../components/GridCards/DashboardGraphModule'
import FilterBreadCrumb from '../../../../components/Filter/FilterBreadCrumb'

const propTypes = {
  data: PropTypes.object,
  filter: PropTypes.object,
  isFetching: PropTypes.bool.isRequired,
  intl: PropTypes.object,
}

const messages = defineMessages({
  premierLeague: {
    id: 'your_fans.tab.media.premier_league',
    defaultMessage: 'Interacting with Premier League',
  },
  useToFollow: {
    id: 'your_fans.tab.media.use_to_follow',
    defaultMessage: 'Which of the following do you use to interact with or follow the Premier League or your favorite Premier League team?',
  },
  activities: {
    id: 'your_fans.tab.media.activities',
    defaultMessage: 'Activities',
  },
  sportWatching: {
    id: 'your_fans.tab.media.watching',
    defaultMessage: 'Sport Watching',
  },
  whileWatching: {
    id: 'your_fans.tab.media.while_watching',
    defaultMessage: 'While Watching Sports',
  },
  howOften: {
    id: 'your_fans.tab.media.how_often',
    defaultMessage: 'How often do you do the following?',
  },
})

const MediaProfileTab = injectIntl(({ intl, filter, data, isFetching }) => (
  <div className={styles.grid}>
    <div className={styles.breadCrumb}>
      <FilterBreadCrumb filter={filter} />
    </div>
    <div className="container">
      <div className="row row-grid">
        <div className="col-6">
          <DashboardGraphModule
            title={intl.formatMessage(messages.premierLeague)}
            description={intl.formatMessage(messages.useToFollow)}
            sideTitle={intl.formatMessage(messages.activities)}
            data={data ? data.media_interacting : null}
            isFetching={isFetching}
            type={MODULE_TYPE.PERCENT}
            propertyTitle={filter.property.name}
            benchmarkTitle={filter.benchmark.name}
          />

          <DashboardGraphModule
            title={intl.formatMessage(messages.sportWatching)}
            data={data ? data.sport_media : null}
            isFetching={isFetching}
            type={MODULE_TYPE.PERCENT}
            propertyTitle={filter.property.name}
            benchmarkTitle={filter.benchmark.name}
          />
        </div>
        <div className="col-6">
          <DashboardGraphModule
            title={intl.formatMessage(messages.whileWatching)}
            description={intl.formatMessage(messages.howOften)}
            data={data ? data.while_watching : null}
            isFetching={isFetching}
            type={MODULE_TYPE.PERCENT}
            propertyTitle={filter.property.name}
            benchmarkTitle={filter.benchmark.name}
          />
        </div>
      </div>
    </div>
  </div>
))

function mapStateToProps(state) {
  const loadingSelector = createLoadingSelector([yourFansActions.GET_HOME_PAGE.SELF])
  const filter = state.filters.filter
  const data = state.pages.yourFans.data
  return { ...state, isFetching: loadingSelector(state), filter, data }
}

MediaProfileTab.propTypes = propTypes
export default connect(mapStateToProps)(MediaProfileTab)
