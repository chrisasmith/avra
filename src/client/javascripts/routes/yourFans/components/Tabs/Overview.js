import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { injectIntl, defineMessages } from 'react-intl'
import styles from '../styles/YourFans.scss'
import * as yourFansActions from '../../../../reducers/pages/yourFansActions'
import { createLoadingSelector } from '../../../../reducers/selectors'
import { MODULE_TYPE } from '../../../../components/GridCards/DashboardGraphModule'
import GridCard, { CARD_TYPE } from '../../../../components/GridCards/GridCard'
import FilterBreadCrumb from '../../../../components/Filter/FilterBreadCrumb'

const propTypes = {
  data: PropTypes.object,
  filter: PropTypes.object,
  isFetching: PropTypes.bool,
  intl: PropTypes.object,
}

const messages = defineMessages({
  fanPopulation: {
    id: 'your_fans.tab.overview.fan_population',
    defaultMessage: 'Fan Population',
  },
  fpByRegion: {
    id: 'your_fans.tab.overview.fp_by_region',
    defaultMessage: 'By Region',
  },
  fpRegion: {
    id: 'your_fans.tab.overview.fp_region',
    defaultMessage: 'Region',
  },
  useSocialPlatform: {
    id: 'your_fans.tab.overview.use_social_platform',
    defaultMessage: 'Do you use any of the following social media platforms?',
  },
  socialPlatform: {
    id: 'your_fans.tab.overview.social_platform',
    defaultMessage: 'Social media platforms',
  },
  favoriteTeam: {
    id: 'your_fans.tab.overview.favorite_team',
    defaultMessage: 'Favorite Team',
  },
  mediaBehavior: {
    id: 'your_fans.tab.overview.media_behavior',
    defaultMessage: 'Media Behavior',
  },
  howOftenDoYouUseSM: {
    id: 'your_fans.tab.overview.how_often_use_sm',
    defaultMessage: 'How often do you use the following media for news about sports?',
  },
  media: {
    id: 'your_fans.tab.overview.media',
    defaultMessage: 'Media',
  },
  fanEngagement: {
    id: 'your_fans.tab.overview.fan_engagement',
    defaultMessage: 'Fan Engagement',
  },
  watchAttendListen: {
    id: 'your_fans.tab.overview.watch_attend_listen',
    defaultMessage: 'Have you watched, attended or listened to events in the last 12 months?',
  },
})

const OverviewTab = injectIntl(({ intl, filter, data, isFetching }) => {
  const leftColumn = [
    {
      type: CARD_TYPE.GRAPH,
      dataKey: 'region',
      cardData: {
        showTotal: true,
        title: intl.formatMessage(messages.fanPopulation),
        description: intl.formatMessage(messages.fpByRegion),
        sideTitle: intl.formatMessage(messages.fpRegion),
        type: MODULE_TYPE.COUNT,
      },
    }, {
      type: CARD_TYPE.GRAPH,
      dataKey: 'social_media',
      cardData: {
        title: intl.formatMessage(messages.useSocialPlatform),
        sideTitle: intl.formatMessage(messages.socialPlatform),
        type: MODULE_TYPE.PERCENT,
      },
    },
  ]

  const rightColumn = [
    {
      type: CARD_TYPE.STRING,
      hide: !data || !data.property || !data.property.team,
      cardData: {
        title: intl.formatMessage(messages.favoriteTeam),
        value: data && data.property && data.property.team ? `${Math.round(data.property.team.favorite)}%` : null,
      },
    }, {
      type: CARD_TYPE.GRAPH,
      dataKey: 'social_media',
      cardData: {
        title: intl.formatMessage(messages.useSocialPlatform),
        sideTitle: intl.formatMessage(messages.socialPlatform),
        type: MODULE_TYPE.PERCENT,
      },
    }, {
      type: CARD_TYPE.GRAPH,
      dataKey: 'media_behavior',
      cardData: {
        title: intl.formatMessage(messages.mediaBehavior),
        description: intl.formatMessage(messages.howOftenDoYouUseSM),
        sideTitle: intl.formatMessage(messages.media),
        type: MODULE_TYPE.PERCENT,
      },
    }, {
      type: CARD_TYPE.GRAPH,
      dataKey: 'fan_engagement',
      cardData: {
        title: intl.formatMessage(messages.fanEngagement),
        description: intl.formatMessage(messages.watchAttendListen),
        type: MODULE_TYPE.PERCENT,
      },
    },
  ]

  return (
    <div className={styles.grid}>
      <div className={styles.breadCrumb}>
        <FilterBreadCrumb filter={filter} />
      </div>
      <div className="container">
        <div className="row row-grid">
          <div className="col-6">
            {leftColumn.map((card, idx) => (
              <GridCard
                key={idx}
                card={card}
                data={data}
                isFetching={isFetching}
                extra={{ propertyTitle: filter.property.name, benchmarkTitle: filter.benchmark.name }}
              />
            ))}
          </div>
          <div className="col-6">
            {rightColumn.map((card, idx) => (
              <GridCard
                key={idx}
                card={card}
                data={data}
                isFetching={isFetching}
                extra={{ propertyTitle: filter.property.name, benchmarkTitle: filter.benchmark.name }}
              />
            ))}
          </div>
        </div>
      </div>
    </div>
  )
})

function mapStateToProps(state) {
  const loadingSelector = createLoadingSelector([yourFansActions.GET_HOME_PAGE.SELF])
  const filter = state.filters.filter
  const data = state.pages.yourFans.data
  return { ...state, isFetching: loadingSelector(state), filter, data }
}

OverviewTab.propTypes = propTypes
export default connect(mapStateToProps)(OverviewTab)
