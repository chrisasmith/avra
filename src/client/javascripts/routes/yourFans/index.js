import YourFans from './components/YourFans'
import InterestProfileTab from './components/Tabs/InterestProfile'
import OverviewTab from './components/Tabs/Overview'
import MediaProfileTab from './components/Tabs/MediaProfile'

module.exports = [{
  path: '/your_fans',
  component: YourFans,
  name: 'ROOT',
  indexRoute: { component: OverviewTab },
}, {
  path: '/your_fans/interest_profile',
  component: YourFans,
  name: 'INTEREST_PROFILE',
  indexRoute: { component: InterestProfileTab },
}, {
  path: '/your_fans/media_profile',
  component: YourFans,
  name: 'MEDIA_PROFILE',
  indexRoute: { component: MediaProfileTab },
}]
