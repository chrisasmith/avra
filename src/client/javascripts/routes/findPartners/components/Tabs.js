import React from 'react'
import { Link } from 'react-router'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import { LinkContainer } from 'react-router-bootstrap'
import { NavDropdown } from 'react-bootstrap'
import { FormattedMessage } from 'react-intl'
import { routes } from '../../../routes'

const propTypes = {
  location: PropTypes.object.isRequired,
  industries: PropTypes.array,
}

const Tabs = ({ location, industries }) => (
  <ul className="nav nav-tabs">
    <li className="nav-item">
      <Link
        className={classnames('nav-link', { active: location.pathname === routes.FINDPARTNERS.ROOT })}
        to={routes.FINDPARTNERS.ROOT}
      >
        <FormattedMessage
          id="find_partners.tabs.overview"
          defaultMessage="Overview"
        />
      </Link>
    </li>
    <NavDropdown title="Industries" active={location.pathname.startsWith(routes.FINDPARTNERS.INDUSTRIES)}>
      {industries && industries.sort((a, b) => a.name.localeCompare(b.name)).map(industry => (
        <LinkContainer to={`${routes.FINDPARTNERS.INDUSTRIES}/${industry.id}`} key={industry.id}>
          <NavDropdown.Item>
            {industry.name}
          </NavDropdown.Item>
        </LinkContainer>
      ))}
    </NavDropdown>
    <li className="nav-item">
      <Link
        className={classnames('nav-link', { active: location.pathname.startsWith(routes.FINDPARTNERS.BRANDS) })}
        to={routes.FINDPARTNERS.BRANDS}
      >
        <FormattedMessage
          id="find_partners.tabs.brands"
          defaultMessage="Brands"
        />
      </Link>
    </li>
    <li className="nav-item">
      <Link
        className={classnames('nav-link', { active: location.pathname.startsWith(routes.FINDPARTNERS.TRENDS) })}
        to={routes.FINDPARTNERS.TRENDS}
      >
        <FormattedMessage
          id="find_partners.tabs.Trends"
          defaultMessage="Trends"
        />
      </Link>
    </li>
  </ul>
)

Tabs.propTypes = propTypes
export default Tabs
