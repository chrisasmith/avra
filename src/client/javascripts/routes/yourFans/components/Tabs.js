import React from 'react'
import { Link } from 'react-router'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import { FormattedMessage } from 'react-intl'
import { routes } from '../../../routes'

const propTypes = {
  location: PropTypes.object.isRequired,
}

const Tabs = ({ location }) => (
  <ul className="nav nav-tabs">
    <li className="nav-item">
      <Link
        className={classnames('nav-link', { active: location.pathname === routes.YOURFANS.ROOT })}
        to={routes.YOURFANS.ROOT}
      >
        <FormattedMessage
          id="your_fans.tabs.overview"
          defaultMessage="Overview"
        />
      </Link>
    </li>
    <li className="nav-item">
      <Link
        className={classnames('nav-link', { active: location.pathname.startsWith(routes.YOURFANS.INTEREST_PROFILE) })}
        to={routes.YOURFANS.INTEREST_PROFILE}
      >
        <FormattedMessage
          id="your_fans.tabs.interest"
          defaultMessage="Interest Profile"
        />
      </Link>
    </li>
    <li className="nav-item">
      <Link
        className={classnames('nav-link', { active: location.pathname.startsWith(routes.YOURFANS.MEDIA_PROFILE) })}
        to={routes.YOURFANS.MEDIA_PROFILE}
      >
        <FormattedMessage
          id="your_fans.tabs.media"
          defaultMessage="Media Profile"
        />
      </Link>
    </li>
    <li className="nav-item">
      <Link
        className={classnames('nav-link', { active: location.pathname.startsWith(routes.YOURFANS.SOCIAL_MEDIA) })}
        to={routes.YOURFANS.SOCIAL_MEDIA}
      >
        <FormattedMessage
          id="your_fans.tabs.social"
          defaultMessage="Social Media"
        />
      </Link>
    </li>
    <li className="nav-item">
      <Link
        className={classnames('nav-link', { active: location.pathname.startsWith(routes.YOURFANS.SEGMENTATION) })}
        to={routes.YOURFANS.SEGMENTATION}
      >
        <FormattedMessage
          id="your_fans.tabs.segment"
          defaultMessage="Segmentation"
        />
      </Link>
    </li>
    <li className="nav-item">
      <Link
        className={classnames('nav-link', { active: location.pathname.startsWith(routes.YOURFANS.TRENDS) })}
        to={routes.YOURFANS.TRENDS}
      >
        <FormattedMessage
          id="your_fans.tabs.trends"
          defaultMessage="Trends"
        />
      </Link>
    </li>
  </ul>
)

Tabs.propTypes = propTypes
export default Tabs
