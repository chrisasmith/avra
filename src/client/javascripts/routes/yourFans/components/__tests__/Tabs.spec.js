import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Tabs from '../Tabs'
import { routes } from '../../../../routes'

describe('Tabs Component', () => {
  it('Should show', () => {
    const wrapper = shallow(<Tabs
      location={{ pathname: '' }}
    />)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('Should show active tab', () => {
    const wrapper = shallow(<Tabs
      location={{ pathname: routes.YOURFANS.INTEREST_PROFILE }}
    />)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
