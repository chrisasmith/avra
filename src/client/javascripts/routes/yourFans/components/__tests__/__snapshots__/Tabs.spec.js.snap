// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Component Should show 1`] = `
<ul
  className="nav nav-tabs"
>
  <li
    className="nav-item"
  >
    <Link
      className="nav-link"
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/your_fans"
    >
      Overview
    </Link>
  </li>
  <li
    className="nav-item"
  >
    <Link
      className="nav-link"
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/your_fans/interest_profile"
    >
      Interest Profile
    </Link>
  </li>
  <li
    className="nav-item"
  >
    <Link
      className="nav-link"
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/your_fans/media_profile"
    >
      Media Profile
    </Link>
  </li>
  <li
    className="nav-item"
  >
    <Link
      className="nav-link"
      onlyActiveOnIndex={false}
      style={Object {}}
    >
      Social Media
    </Link>
  </li>
  <li
    className="nav-item"
  >
    <Link
      className="nav-link"
      onlyActiveOnIndex={false}
      style={Object {}}
    >
      Segmentation
    </Link>
  </li>
  <li
    className="nav-item"
  >
    <Link
      className="nav-link"
      onlyActiveOnIndex={false}
      style={Object {}}
    >
      Trends
    </Link>
  </li>
</ul>
`;

exports[`Tabs Component Should show active tab 1`] = `
<ul
  className="nav nav-tabs"
>
  <li
    className="nav-item"
  >
    <Link
      className="nav-link"
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/your_fans"
    >
      Overview
    </Link>
  </li>
  <li
    className="nav-item"
  >
    <Link
      className="nav-link active"
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/your_fans/interest_profile"
    >
      Interest Profile
    </Link>
  </li>
  <li
    className="nav-item"
  >
    <Link
      className="nav-link"
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/your_fans/media_profile"
    >
      Media Profile
    </Link>
  </li>
  <li
    className="nav-item"
  >
    <Link
      className="nav-link"
      onlyActiveOnIndex={false}
      style={Object {}}
    >
      Social Media
    </Link>
  </li>
  <li
    className="nav-item"
  >
    <Link
      className="nav-link"
      onlyActiveOnIndex={false}
      style={Object {}}
    >
      Segmentation
    </Link>
  </li>
  <li
    className="nav-item"
  >
    <Link
      className="nav-link"
      onlyActiveOnIndex={false}
      style={Object {}}
    >
      Trends
    </Link>
  </li>
</ul>
`;
